### PR TITLE
Bug 2073220: routes/status resources can leak sensitive data

### PR DESCRIPTION
**Pull request by aojea**
Fixes: https://bugzilla.redhat.com/show_bug.cgi?id=2073220

### CODE REVIEW

**Comment by tkashem** [comment]
`TestGetAuditPolicy` is failing, i think you will need to add the same in the test files as well

**Comment by tkashem** [comment]
/retitle Bug 2073220: routes/status resources can leak sensitive data

**Comment by tkashem** [comment]
> /retest-required

@aojea I think you will need to add the change in the test data as well

https://github.com/openshift/library-go/blob/330d9097d0f08c86aedafc8ec350b25d4a9b4f68/pkg/operator/apiserver/audit/testdata/writerequestbodies.yaml#L35

maybe here as well? 
https://github.com/openshift/library-go/blob/330d9097d0f08c86aedafc8ec350b25d4a9b4f68/pkg/operator/apiserver/audit/testdata/allrequestbodies.yaml#L35

**Comment by tkashem** [comment]
found another one 
https://github.com/openshift/library-go/blob/330d9097d0f08c86aedafc8ec350b25d4a9b4f68/pkg/operator/apiserver/audit/testdata/audit-policies-cm-scenario-1.yaml#L124

**Comment by aojea** [comment]
I don't have a clue what this fails, if I run it locally the failing test looks unrelated

```
 /usr/local/go/bin/go test -timeout 120s -run ^Test_newFromConfigMapData_slow_sigstore$ github.com/openshift/library-go/pkg/verify -v

=== RUN   Test_newFromConfigMapData_slow_sigstore
    /home/aojea/go/src/github.com/openshift/library-go/pkg/verify/configmap_test.go:153: expected ^[[][0-9TZ:-]*: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, [0-9TZ:-]*: context deadline exceeded]$, got [2022-09-01T19:39:04+02:00: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, 2022-09-01T19:39:04+02:00: context deadline exceeded]
--- FAIL: Test_newFromConfigMapData_slow_sigstore (0.10s)
FAIL
FAIL	github.com/openshift/library-go/pkg/verify	0.114s
```

File: pkg/verify/configmap_test.go [hunk 1]
```diff
@@ -148,7 +148,7 @@ func Test_newFromConfigMapData_slow_sigstore(t *testing.T) {
 	}
 
 	wrapped := errors.Unwrap(err)
-	expected = regexp.MustCompile(`^[[][0-9TZ:-]*: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, [0-9TZ:-]*: context deadline exceeded]$`)
+	expected = regexp.MustCompile(`^[[][0-9TZ:+-]*: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, [0-9TZ:+-]*: context deadline exceeded]$`)
```
Comment (aojea) [inline pkg/verify/configmap_test.go:151]:
```
--- FAIL: Test_newFromConfigMapData_slow_sigstore (0.10s)
    configmap_test.go:153: expected ^[[][0-9TZ:-]*: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, [0-9TZ:-]*: context deadline exceeded]$, got [2022-09-02T10:05:33+02:00: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, 2022-09-02T10:05:33+02:00: context deadline exceeded]
FAIL
```
:/

**Review by soltysh — APPROVED** [review]
/approve

**Comment by tkashem** [comment]
```
$ go test -v -race -count=1 ./pkg/verify -run=Test_newFromConfigMapData_slow_sigstore
=== RUN   Test_newFromConfigMapData_slow_sigstore
    configmap_test.go:153: [2022-09-02T07:57:51-04:00: Get "https://example.com/signatures/sha256=123/signature-1": context deadline exceeded, 2022-09-02T07:57:51-04:00: context deadline exceeded]
--- PASS: Test_newFromConfigMapData_slow_sigstore (0.10s)
PASS
ok  	github.com/openshift/library-go/pkg/verify	0.183s
```

If I run it locally, it passes for me, (i am logging the error ^), and my go version is `go version go1.19 linux/amd64`
and the timestamp i see is `2022-09-02T07:57:51-04:00` ( `-04:00` is my time zone)

This line is creating the error with timestamp `time.Now().Format(time.RFC3339)`
https://github.com/openshift/library-go/blob/330d9097d0f08c86aedafc8ec350b25d4a9b4f68/pkg/verify/verify.go#L192

`time.RFC3339` translates to `2006-01-02T15:04:05Z07:00`

maybe in ci, this test is running in a `+00:00` timezone now? @aojea the change looks good to me

**Pull request merged**